source/exec/iexecprovider.d

```diff
@@ -8,5 +8,5 @@ import std.typecons: Tuple;
 +/
 interface IExecProvider
 {
-	Tuple!(string, "output", bool, "success") compileAndExecute(string source);
+	Tuple!(string, "output", bool, "success") compileAndExecute(string source, string compiler = "dmd");
```

Comment:
We could also think about making `compiler` an `enum` and do the validation in the REST API.